### PR TITLE
[INTERNAL] specVersion 2.2 and taskUtil/middlewareUtil documentation

### PR DESCRIPTION
**Pull request by RandomByte**

### CODE REVIEW

File: docs/pages/extensibility/CustomServerMiddleware.md [hunk 1]
```diff
@@ -137,3 +139,9 @@ module.exports = function({resources, options}) {
 ````
 
 Live demo of the above example: https://github.com/SAP/openui5-sample-app/tree/demo-server-middleware-extensibility
+
+## Helper Class `MiddlewareUtil`
+
+Custom middleware defining [Specification Version](../Configuration.md#specification-versions) 2.0 or higher have access to an interface of a [MiddlewareUtil](https://sap.github.io/ui5-tooling/api/module-@ui5_server.middleware.MiddlewareUtil.html) instance.
+
+In this case, a `middlewareUtil` object is provided as part of the custom middleware's [parameters](#custom-middleware-implementation).  Depending on the Specification Version of the custom middleware, a set of helper functions is available to the implementation. The lowest required Specification Version for every function is listed in the [MiddlewareUtil API reference](https://sap.github.io/ui5-tooling/api/module-@ui5_server.middleware.MiddlewareUtil.html).
```
Comment (KlattG) [inline docs/pages/extensibility/CustomServerMiddleware.md:147]:
```suggestion
In this case, a `middlewareUtil` object is provided as part of the custom middleware's [parameters](#custom-middleware-implementation).  Depending on the specification version of the custom middleware, a set of helper functions is available to the implementation. The lowest required specification version for every function is listed in the [MiddlewareUtil API reference](https://sap.github.io/ui5-tooling/api/module-@ui5_server.middleware.MiddlewareUtil.html).
```

File: docs/pages/extensibility/CustomTasks.md [hunk 2]
```diff
@@ -136,3 +138,10 @@ module.exports = async function({workspace, dependencies, options}) {
 
 !!! warning
     Depending on your project setup, the UI5 Tooling tends to have lots of open files at the same time during a build. To prevent errors like `EMFILE: too many open files`, we urge custom task implementations to use the [graceful-fs](https://github.com/isaacs/node-graceful-fs#readme) module as a drop-in replacement for the native `fs` module.
+
+
+## Helper Class `TaskUtil`
+
+Custom tasks defining [Specification Version](../Configuration.md#specification-versions) 2.2 or higher have access to an interface of a [TaskUtil](https://sap.github.io/ui5-tooling/api/module-@ui5_builder.tasks.TaskUtil.html) instance.
+
+In this case, a `taskUtil` object is provided as part of the custom task's [parameters](#task-implementation).  Depending on the Specification Version of the custom task, a set of helper functions is available to the implementation. The lowest required Specification Version for every function is listed in the [TaskUtil API reference](https://sap.github.io/ui5-tooling/api/module-@ui5_builder.tasks.TaskUtil.html).
```
Comment (KlattG) [inline docs/pages/extensibility/CustomTasks.md:147]: see my comment above

File: docs/pages/extensibility/CustomServerMiddleware.md (hunk 1 above)
Comment (RandomByte) [reply docs/pages/extensibility/CustomServerMiddleware.md:147]: Done

File: docs/pages/extensibility/CustomTasks.md (hunk 2 above)
Comment (RandomByte) [reply docs/pages/extensibility/CustomTasks.md:147]: Done

Comment (ecker) [reply docs/pages/extensibility/CustomTasks.md:147]: Looks like the two terms are still uppercase?

File: docs/pages/Configuration.md [hunk 3]
```diff
@@ -461,12 +461,21 @@ Unless otherwise noted in the table below, the UI5 Tooling modules are backward
...
 **2.0** | v2.0.0+
 **1.1** | v1.13.0+
 **1.0** | v1.0.0+
 **0.1** | v0.0.1+
 
+### Specification Version 2.2
+
+**Features:**
+
+- TODO
```
Comment (ecker) [inline docs/pages/Configuration.md:475]: Likely marked by intention as an open issue?

File: docs/pages/extensibility/CustomTasks.md (hunk 2 above)
Comment (RandomByte) [reply docs/pages/extensibility/CustomTasks.md:147]: The link one reads "Specification Version 2.2" and therefore refers to a specific version. In that case it should be uppercased. Also see https://github.com/SAP/ui5-builder/pull/480#discussion_r466181782

The two occasions in the second paragraph are now lowercase.

File: docs/pages/Configuration.md (hunk 3 above)
Comment (RandomByte) [reply docs/pages/Configuration.md:475]: Indeed, I'll take care of it. Thanks!

Comment (RandomByte) [reply docs/pages/Configuration.md]: Done in 8cedb82

File: docs/pages/extensibility/CustomServerMiddleware.md [hunk 4]
```diff
@@ -137,3 +139,9 @@ module.exports = function({resources, options}) {
 ````
 
 Live demo of the above example: https://github.com/SAP/openui5-sample-app/tree/demo-server-middleware-extensibility
+
+## Helper Class `MiddlewareUtil`
+
+Custom middleware defining [Specification Version](../Configuration.md#specification-versions) 2.0 or higher have access to an interface of a [MiddlewareUtil](https://sap.github.io/ui5-tooling/api/module-@ui5_server.middleware.MiddlewareUtil.html) instance.
+
+In this case, a `middlewareUtil` object is provided as part of the custom middleware's [parameters](#custom-middleware-implementation).  Depending on the specification version of the custom middleware, a set of helper functions is available to the implementation. The lowest required specification version for every function is listed in the [MiddlewareUtil API reference](https://sap.github.io/ui5-tooling/api/module-@ui5_server.middleware.MiddlewareUtil.html).
```
Comment (tobiasso85) [inline docs/pages/extensibility/CustomServerMiddleware.md:147]:
```suggestion
In this case, a `middlewareUtil` object is provided as part of the custom middleware's [parameters](#custom-middleware-implementation). Depending on the specification version of the custom middleware, a set of helper functions is available to the implementation. The lowest required specification version for every function is listed in the [MiddlewareUtil API reference](https://sap.github.io/ui5-tooling/api/module-@ui5_server.middleware.MiddlewareUtil.html).
```

File: docs/pages/extensibility/CustomTasks.md (hunk 2 above)
Comment (tobiasso85) [inline docs/pages/extensibility/CustomTasks.md:147]:
```suggestion
In this case, a `taskUtil` object is provided as part of the custom task's [parameters](#task-implementation). Depending on the Specification Version of the custom task, a set of helper functions is available to the implementation. The lowest required Specification Version for every function is listed in the [TaskUtil API reference](https://sap.github.io/ui5-tooling/api/module-@ui5_builder.tasks.TaskUtil.html).
```

File: docs/pages/extensibility/CustomServerMiddleware.md (hunk 4 above)
Comment (RandomByte) [reply docs/pages/extensibility/CustomServerMiddleware.md]: Done

File: docs/pages/extensibility/CustomTasks.md (hunk 2 above)
Comment (RandomByte) [reply docs/pages/extensibility/CustomTasks.md]: Done

File: docs/pages/extensibility/CustomTasks.md [hunk 5]
```diff
@@ -136,3 +138,10 @@ module.exports = async function({workspace, dependencies, options}) {
 
 !!! warning
     Depending on your project setup, the UI5 Tooling tends to have lots of open files at the same time during a build. To prevent errors like `EMFILE: too many open files`, we urge custom task implementations to use the [graceful-fs](https://github.com/isaacs/node-graceful-fs#readme) module as a drop-in replacement for the native `fs` module.
+
+
+## Helper Class `TaskUtil`
+
+Custom tasks defining [Specification Version](../Configuration.md#specification-versions) 2.2 or higher have access to an interface of a [TaskUtil](https://sap.github.io/ui5-tooling/api/module-@ui5_builder.tasks.TaskUtil.html) instance.
+
+In this case, a `taskUtil` object is provided as part of the custom task's [parameters](#task-implementation). Depending on the Specification Version of the custom task, a set of helper functions is available to the implementation. The lowest required Specification Version for every function is listed in the [TaskUtil API reference](https://sap.github.io/ui5-tooling/api/module-@ui5_builder.tasks.TaskUtil.html).
```
Comment (KlattG) [inline docs/pages/extensibility/CustomTasks.md:147]:
```suggestion
In this case, a `taskUtil` object is provided as a part of the custom task's [parameters](#task-implementation). Depending on the specification version of the custom task, a set of helper functions is available to the implementation. The lowest required specification version for every function is listed in the [TaskUtil API reference](https://sap.github.io/ui5-tooling/api/module-@ui5_builder.tasks.TaskUtil.html).
```

File: docs/pages/extensibility/CustomServerMiddleware.md [hunk 6]
```diff
@@ -137,3 +139,9 @@ module.exports = function({resources, options}) {
 ````
 
 Live demo of the above example: https://github.com/SAP/openui5-sample-app/tree/demo-server-middleware-extensibility
+
+## Helper Class `MiddlewareUtil`
+
+Custom middleware defining [Specification Version](../Configuration.md#specification-versions) 2.0 or higher have access to an interface of a [MiddlewareUtil](https://sap.github.io/ui5-tooling/api/module-@ui5_server.middleware.MiddlewareUtil.html) instance.
+
+In this case, a `middlewareUtil` object is provided as part of the custom middleware's [parameters](#custom-middleware-implementation). Depending on the specification version of the custom middleware, a set of helper functions is available to the implementation. The lowest required specification version for every function is listed in the [MiddlewareUtil API reference](https://sap.github.io/ui5-tooling/api/module-@ui5_server.middleware.MiddlewareUtil.html).
```
Comment (KlattG) [inline docs/pages/extensibility/CustomServerMiddleware.md:147]:
```suggestion
In this case, a `middlewareUtil` object is provided as a part of the custom middleware's [parameters](#custom-middleware-implementation). Depending on the specification version of the custom middleware, a set of helper functions is available to the implementation. The lowest required specification version for every function is listed in the [MiddlewareUtil API reference](https://sap.github.io/ui5-tooling/api/module-@ui5_server.middleware.MiddlewareUtil.html).
```

Comment (RandomByte) [reply docs/pages/extensibility/CustomServerMiddleware.md]: Done

File: docs/pages/extensibility/CustomTasks.md (hunk 5 above)
Comment (RandomByte) [reply docs/pages/extensibility/CustomTasks.md]: Done

**Pull request merged**